crates/cheatcodes/src/utils.rs

```diff
@@ -202,9 +209,32 @@ fn create_wallet(private_key: &U256, label: Option<&str>, state: &mut Cheatcodes
 }
 
 fn encode_vrs(sig: alloy_primitives::Signature) -> Vec<u8> {
+    // Retrieve v, r and s from signature.
+    let v = U256::from(sig.v().y_parity_byte_non_eip155().unwrap_or(sig.v().y_parity_byte()));
+    let r = B256::from(sig.r());
+    let s = B256::from(sig.s());
+    (v, r, s).abi_encode()
+}
+
+fn encode_rvs(sig: alloy_primitives::Signature) -> Vec<u8> {
+    // Retrieve v, r and s from signature.
     let v = sig.v().y_parity_byte_non_eip155().unwrap_or(sig.v().y_parity_byte());
+    let r = B256::from(sig.r());
+    let mut vs = B256::from(sig.s());
```

Comment:
We can keep `s` as a `U256` and use `sig.y_parity()` to simplify the implementation: `s.set_bit(255, sig.y_parity())`

Reply:
done in [chore: update encode_compact_signature impl](https://github.com/foundry-rs/foundry/pull/8538/commits/889007a39ac27ba025223e9c8a2bbd9eb40840eb)